core/storage/topic/gae_models.py

```diff
@@ -102,6 +102,18 @@ def _trusted_commit(
         topic_commit_log_entry.topic_id = self.id
         topic_commit_log_entry.put()
 
+    @classmethod
+    def get_by_name(cls, topic_name):
+        """Gets TopicModel by topic_name.
+
+        Args:
+            topic_name: str. The name of the topic.
+
+        Returns:
+            TopicModel. The summary model of the question.
+        """
+        return TopicModel.query().filter(cls.name == topic_name).fetch()
```

Comment:
This should return only a single topic instance.

There should also be checks prior to saving any topic that no two topic names in the datastore can coincide.

Reply:
Done and updated ```save_new_story```